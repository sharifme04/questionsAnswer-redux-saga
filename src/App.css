.custom-answer {
  border: 1px solid #d9d9d9;
  margin-bottom: 0.1rem;
  padding: 0.5rem
}

.card {
  border: 1px solid #d9d9d9;
  padding: 10px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.correct-color {
  background-color: #80ffbf
}

.incorrect-color {
  background-color: #ff9999
}

.custom-spin{
  width: 75px;
  height: 75px;
}
