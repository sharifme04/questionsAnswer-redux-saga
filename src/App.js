import React, { useState, useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { BrowserRouter, Route } from "react-router-dom";
import "./App.css";
import { questionsRequest } from "./app/redux/actions/questions";
import QuestionsList from "./app/components/QuestionsList";
import Results from "./app/components/Results";

function App() {
  const dispatch = useDispatch();
  const [amount] = useState(10);
  const questions = useSelector(state => state.questions);
  let [selectedCorrect, setSelectedCorrect] = useState([]);
  let [selectedIncorrect, setSelectedIncorrect] = useState([]);
  let [correctKey, setCorrectKey] = useState([]);
  let [incorrectKey, setIncorrectKey] = useState([]);

  useEffect(() => {
    dispatch(questionsRequest(amount));
  }, [amount, dispatch]);

  const handleClick = (answer, i, question, correctAnswer) => {
    if (
      answer === questions.questions.results[i].correct_answer &&
      !correctKey.includes(i) &&
      !selectedCorrect.some(el => el[i] === answer)
    ) {
      correctKey = [...correctKey, i];
      selectedCorrect = [...selectedCorrect, { [i]: answer }];
      setSelectedCorrect(selectedCorrect);
      setCorrectKey(correctKey);
    }
    if (
      answer !== questions.questions.results[i].correct_answer &&
      !incorrectKey.includes(i) &&
      !correctKey.includes(i)
    ) {
      selectedIncorrect = [...selectedIncorrect, { [i]: answer }];
      selectedCorrect = [...selectedCorrect, { [i]: correctAnswer }];
      incorrectKey = [...incorrectKey, i];
      setSelectedIncorrect(selectedIncorrect);
      setIncorrectKey(incorrectKey);
      setSelectedCorrect(selectedCorrect);
    }
  };

  const updateResults =
    questions &&
    questions.questions &&
    questions.questions.results &&
    questions.questions.results.map(e => ({
      ...e,
      allAnswers: [...e.incorrect_answers, e.correct_answer]
    }));

  return (
    <BrowserRouter>
      <div className="container-fluid mt-1">
        <Route
          exact
          path="/"
          render={props => (
            <QuestionsList
              questions={questions && questions}
              handleClick={handleClick}
              updateResults={updateResults && updateResults}
              totalQuestions={
                updateResults && updateResults.length && updateResults.length
              }
              selectedCorrect={selectedCorrect}
              selectedIncorrect={selectedIncorrect}
              {...props}
            />
          )}
        />
        <Route
          exact
          path="/results"
          render={props => (
            <Results
              {...props}
              updateResults={updateResults && updateResults}
              selectedCorrect={selectedCorrect}
              selectedIncorrect={selectedIncorrect}
            />
          )}
        />
      </div>
    </BrowserRouter>
  );
}

export default App;
