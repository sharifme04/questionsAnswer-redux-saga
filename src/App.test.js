import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import App from "./App";

const mockStore = configureMockStore();
const store = mockStore({});
const fakeQuestions = {
  isLoading: false,
  questions: {
    response_code: 0,
    results: [
      {
        category: "Entertainment: Film",
        type: "multiple",
        difficulty: "easy",
        question:
          "Who played Deputy Marshal Samuel Gerard in the 1993 film &quot;The Fugitive&quot;?",
        correct_answer: "Tommy Lee Jones",
        incorrect_answers: ["Harrison Ford", "Harvey Keitel", "Martin Landau"]
      }
    ]
  }
};

test("should render initial layout", () => {
  const component = shallow(
    <Provider store={store}>
      <MemoryRouter initialEntries={["/"]}>
        <App />
      </MemoryRouter>
    </Provider>
  );
  expect(component.exists()).toBe(true);
});

test("snapshot data renders", () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter initialEntries={["/"]}>
        <App questions={fakeQuestions} />
      </MemoryRouter>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot renders", () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter initialEntries={["/"]}>
        <App />
      </MemoryRouter>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
