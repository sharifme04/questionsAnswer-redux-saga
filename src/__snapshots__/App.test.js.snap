// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot data renders 1`] = `
<div
  className="container-fluid mt-1"
>
  <div
    className="mt-1"
  >
    <h2>
      Deevio questions app
    </h2>
    <hr />
    <div
      className="row justify-content-center"
    >
      <h5>
         Questions are not available
      </h5>
      <div
        className="mt-3 mb-4"
      >
        <hr />
      </div>
    </div>
  </div>
</div>
`;

exports[`snapshot renders 1`] = `
<div
  className="container-fluid mt-1"
>
  <div
    className="mt-1"
  >
    <h2>
      Deevio questions app
    </h2>
    <hr />
    <div
      className="row justify-content-center"
    >
      <h5>
         Questions are not available
      </h5>
      <div
        className="mt-3 mb-4"
      >
        <hr />
      </div>
    </div>
  </div>
</div>
`;
